src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -273,6 +278,58 @@ EmberAfNetworkCommissioningError OnEnableNetworkCommandCallbackInternal(app::Com
     return err;
 }
 
+static void ThreadScanNetworksResultCallback(chip::DeviceLayer::ConnectivityManager::ThreadDiscoveryResult * result)
+{
+    if (result == nullptr)
+    {
+        return;
+    }
+
+    ChipLogProgress(DeviceLayer, "ThreadDiscoveryResult for scanned network index = %d:", sThreadNetworksScanned);
+    ChipLogProgress(DeviceLayer, "- discoveryResponseVersion: %d", result->discoveryResponseVersion);
+    ChipLogProgress(DeviceLayer, "- discoveryResponseNativeCommissionerFlag: %d", result->discoveryResponseNativeCommissionerFlag);
+    ChipLogProgress(DeviceLayer, "- extendedPanId: %02X%02X%02X%02X%02X%02X%02X%02X", result->extendedPanId[0],
+                    result->extendedPanId[1], result->extendedPanId[2], result->extendedPanId[3], result->extendedPanId[4],
+                    result->extendedPanId[5], result->extendedPanId[6], result->extendedPanId[7]);
+    ChipLogProgress(DeviceLayer, "- networkName: %s", result->networkName);
+    ChipLogProgress(DeviceLayer, "- steeringDataLength: %s", result->steeringDataLength);
+    ChipLogProgress(DeviceLayer, "- steeringData: %02X%02X%02X%02X%02X%02X%02X%02X%02X%02X%02X%02X%02X%02X%02X%02X",
+                    result->steeringData[0], result->steeringData[1], result->steeringData[2], result->steeringData[3],
+                    result->steeringData[4], result->steeringData[5], result->steeringData[6], result->steeringData[7],
+                    result->steeringData[8], result->steeringData[9], result->steeringData[10], result->steeringData[11],
+                    result->steeringData[12], result->steeringData[13], result->steeringData[14], result->steeringData[15]);
+    ChipLogProgress(DeviceLayer, "- joinerUdpPort: %d", result->joinerUdpPort);
+
+    if (sThreadNetworksScanned >= kMaxScannedThreadNetworks)
+    {
+        ChipLogError(DeviceLayer, "Scanned network hasn't been added to the list due to lack of free space.");
+        return;
+    }
+    memcpy(&sThreadScanResults[sThreadNetworksScanned], result,
```

Comment:
sThreadScanResults[sThreadNetworksScanned] = *result ?

Reply:
I changed it to result assignment, as it is now reference on an object.